tuna/export_db.py

```diff
@@ -95,237 +105,240 @@ def parse_args():
-  elif db_type == DB_Type.KERN_DB:
-    extension = '.kdb'
-  else:
-    extension = ".db"
+    LOGGER.info("MIOpen fdb: %s, cfg: %s, slv: %s", fastest_entry.fdb_key,
+                fastest_entry.config, ID_SOLVER_MAP[fastest_entry.solver])
 
-  final_name = f"{final_name}{extension}"
+  LOGGER.warning("Total number of entries in Find DB: %s", num_fdb_entries)
 
-  return final_name
+  return miopen_fdb
 
 
 def write_fdb(arch, num_cu, ocl, find_db, filename=None):
   """
   Serialize find_db map to plain text file in MIOpen format
   """
-  _, id_solver_map_h = get_id_solvers()
   file_name = get_filename(arch, num_cu, filename, ocl, DB_Type.FIND_DB)
-  FDBRecord = namedtuple('FDBRecord',
-                         'alg_lib solver_id kernel_time workspace_sz')
 
   with open(file_name, 'w') as out:  # pylint: disable=unspecified-encoding
     for key, solvers in sorted(find_db.items(), key=lambda kv: kv[0]):
-      solvers.sort(key=lambda x: float(x[2]))
+      solvers.sort(key=lambda x: float(x.kernel_time))
       lst = []
       # for alg_lib, solver_id, kernel_time, workspace_sz in solvers:
       for rec in solvers:
-        rec = FDBRecord(*rec)
         # pylint: disable-next=consider-using-f-string ; more reable
-        lst.append('{alg}:{},{},{},{alg},{}'.format(
-            id_solver_map_h[rec.solver_id],
-            rec.kernel_time,
-            rec.workspace_sz,
-            'not used',
-            alg=rec.alg_lib))
+        lst.append('{alg}:{},{},{},{alg},{}'.format(ID_SOLVER_MAP[rec.solver],
+                                                    rec.kernel_time,
+                                                    rec.workspace_sz,
+                                                    'not used',
+                                                    alg=rec.alg_lib))
       out.write(f"{key}={';'.join(lst)}\n")
   return file_name
 
 
-def build_miopen_fdb(find_key_alg_lists):
-  """ create miopen find db object for export
+def export_fdb(dbt, args):
+  """Function to export find_db to txt file
+  """
+  query = get_fdb_query(dbt, args)
+  fdb_alg_lists = get_fdb_alg_lists(query)
+  miopen_fdb = build_miopen_fdb(fdb_alg_lists)
+
+  return write_fdb(args.arch, args.num_cu, args.opencl, miopen_fdb,
+                   args.filename)
+
+
+def build_miopen_kdb(dbt, find_db):
+  """ create miopen kernel db object for export
   """
   num_fdb_entries = 0
-  miopen_fdb = OrderedDict()
-  for fdb_key_alg, items in find_key_alg_lists.items():
-    fdb_key = fdb_key_alg[0]
-    num_fdb_entries += 1
-    #pick fastest solver for each algorithm
-    items.sort(key=lambda x: float(x[2]))
-    fastest_entry = items[0]
-    lst = miopen_fdb.get(fdb_key)
-    if not lst:
-      miopen_fdb[fdb_key] = [fastest_entry]
-    else:
-      lst.append(fastest_entry)
+  num_kdb_blobs = 0
+  kern_db = []
+  for _, entries in find_db.items():
+    for fdb_entry in entries:
+      num_fdb_entries += 1
+      with DbSession() as session:
```

Comment:
Please encompass all the loops in this with DbSession statement otherwise we are getting a new db session for each i,j in this double loop and we dont want to me making that many db connections.